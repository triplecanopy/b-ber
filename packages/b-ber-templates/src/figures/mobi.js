import { media, unsupported, figure } from './helpers'

const mobi = {
  portrait: data => figure({ data, ratioName: 'portrait', linkImages: false }),
  landscape: data =>
    figure({ data, ratioName: 'landscape', linkImages: false }),
  'portrait-high': data =>
    figure({ data, ratioName: 'portrait-high', linkImages: false }),
  square: data => figure({ data, ratioName: 'square', linkImages: false }),
  audio: data => media({ ...data, applyInlineClasses: true }),
  video: data => media({ ...data, applyInlineClasses: true }),
  iframe: data => unsupported({ data, applyInlineClasses: true }),
  vimeo: data => unsupported({ data, applyInlineClasses: true }),
}

export default mobi
