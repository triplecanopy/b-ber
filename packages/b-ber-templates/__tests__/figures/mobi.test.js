import fs from 'fs-extra'
import 'image-size'
import mobi from '../../src/figures/mobi'

jest.mock('@canopycanopycanopy/b-ber-lib/State')
jest.mock('image-size', () => jest.fn(() => ({ width: 10, height: 10 })))

afterAll(() => fs.remove('_project'))

describe('templates.mobi', () => {
  test('creates the markup', () => {
    const data = {
      inline: true,
      classes: 'test',
      id: 1,
      ref: 1,
      alt: 'test',
      source: 'test.jpg',
      caption: 'test',
      attrString: 'autoplay="false"',
      sourceElements: '<source />',
      mediaType: 'audio-or-video',
      poster: 'test.jpg',
    }

    const dataFiguresPage = Object.assign({}, data, { inline: false })
    const dataPagebreakBefore = Object.assign({}, data, {
      classes: 'break-before',
    })

    expect(mobi.portrait(data)).toMatchSnapshot()
    expect(mobi.landscape(data)).toMatchSnapshot()
    expect(mobi['portrait-high'](data)).toMatchSnapshot()
    expect(mobi.square(data)).toMatchSnapshot()
    expect(mobi.audio(data)).toMatchSnapshot()
    expect(mobi.video(data)).toMatchSnapshot()
    expect(mobi.iframe(data)).toMatchSnapshot()
    expect(mobi.vimeo(data)).toMatchSnapshot()

    expect(mobi.portrait(dataFiguresPage)).toMatchSnapshot()
    expect(mobi.landscape(dataFiguresPage)).toMatchSnapshot()
    expect(mobi['portrait-high'](dataFiguresPage)).toMatchSnapshot()
    expect(mobi.square(dataFiguresPage)).toMatchSnapshot()
    expect(mobi.audio(dataFiguresPage)).toMatchSnapshot()
    expect(mobi.video(dataFiguresPage)).toMatchSnapshot()
    expect(mobi.iframe(dataFiguresPage)).toMatchSnapshot()
    expect(mobi.vimeo(dataFiguresPage)).toMatchSnapshot()

    expect(mobi.portrait(dataPagebreakBefore)).toMatchSnapshot()
    expect(mobi.landscape(dataPagebreakBefore)).toMatchSnapshot()
    expect(mobi['portrait-high'](dataPagebreakBefore)).toMatchSnapshot()
    expect(mobi.square(dataPagebreakBefore)).toMatchSnapshot()
    expect(mobi.audio(dataPagebreakBefore)).toMatchSnapshot()
    expect(mobi.video(dataPagebreakBefore)).toMatchSnapshot()
    expect(mobi.iframe(dataPagebreakBefore)).toMatchSnapshot()
    expect(mobi.vimeo(dataPagebreakBefore)).toMatchSnapshot()
  })
})
