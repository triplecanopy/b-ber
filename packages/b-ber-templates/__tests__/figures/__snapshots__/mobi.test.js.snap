// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates.mobi creates the markup 1`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--portrait test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <span>
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test.jpg\\"/>
    </span>
      <div class=\\"figcaption\\">
        <p class=\\"small\\">
          test
          
          
        </p>
      </div>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.mobi creates the markup 2`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--landscape test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <span>
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test.jpg\\"/>
    </span>
      <div class=\\"figcaption\\">
        <p class=\\"small\\">
          test
          
          
        </p>
      </div>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.mobi creates the markup 3`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--portrait-high test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <span>
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test.jpg\\"/>
    </span>
      <div class=\\"figcaption\\">
        <p class=\\"small\\">
          test
          
          
        </p>
      </div>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.mobi creates the markup 4`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <span>
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test.jpg\\"/>
    </span>
      <div class=\\"figcaption\\">
        <p class=\\"small\\">
          test
          
          
        </p>
      </div>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.mobi creates the markup 5`] = `
"
    
      <div class=\\"figure__large figure__large--media figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.mobi creates the markup 6`] = `
"
    
      <div class=\\"figure__large figure__large--media figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.mobi creates the markup 7`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"square\\" alt=\\"Media Fallback Image\\" src=\\"../images/test.jpg\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">Your device does not support embedded media.</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.mobi creates the markup 8`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"square\\" alt=\\"Media Fallback Image\\" src=\\"../images/test.jpg\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">Your device does not support embedded media.</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.mobi creates the markup 9`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--portrait\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <span>
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test.jpg\\"/>
    </span>
      <div class=\\"figcaption\\">
        <p class=\\"small\\">
          test
          <br/>
          <a href=\\"1.xhtml#ref1\\">Return</a>
        </p>
      </div>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.mobi creates the markup 10`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--landscape\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <span>
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test.jpg\\"/>
    </span>
      <div class=\\"figcaption\\">
        <p class=\\"small\\">
          test
          <br/>
          <a href=\\"1.xhtml#ref1\\">Return</a>
        </p>
      </div>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.mobi creates the markup 11`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--portrait-high\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <span>
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test.jpg\\"/>
    </span>
      <div class=\\"figcaption\\">
        <p class=\\"small\\">
          test
          <br/>
          <a href=\\"1.xhtml#ref1\\">Return</a>
        </p>
      </div>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.mobi creates the markup 12`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--square\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <span>
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test.jpg\\"/>
    </span>
      <div class=\\"figcaption\\">
        <p class=\\"small\\">
          test
          <br/>
          <a href=\\"1.xhtml#ref1\\">Return</a>
        </p>
      </div>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.mobi creates the markup 13`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--media figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    </section>
    "
`;

exports[`templates.mobi creates the markup 14`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--media figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    </section>
    "
`;

exports[`templates.mobi creates the markup 15`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--square\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <a href=\\"1.xhtml#ref1\\">
      <img class=\\"square\\" alt=\\"Media Fallback Image\\" src=\\"../images/test.jpg\\"/>
    </a>
    <div class=\\"figcaption\\">
      <p class=\\"small\\">Your device does not support embedded media.</p>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.mobi creates the markup 16`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--square\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <a href=\\"1.xhtml#ref1\\">
      <img class=\\"square\\" alt=\\"Media Fallback Image\\" src=\\"../images/test.jpg\\"/>
    </a>
    <div class=\\"figcaption\\">
      <p class=\\"small\\">Your device does not support embedded media.</p>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.mobi creates the markup 17`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--portrait break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <span>
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test.jpg\\"/>
    </span>
      <div class=\\"figcaption\\">
        <p class=\\"small\\">
          test
          
          
        </p>
      </div>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.mobi creates the markup 18`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--landscape break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <span>
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test.jpg\\"/>
    </span>
      <div class=\\"figcaption\\">
        <p class=\\"small\\">
          test
          
          
        </p>
      </div>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.mobi creates the markup 19`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--portrait-high break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <span>
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test.jpg\\"/>
    </span>
      <div class=\\"figcaption\\">
        <p class=\\"small\\">
          test
          
          
        </p>
      </div>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.mobi creates the markup 20`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--square break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <span>
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test.jpg\\"/>
    </span>
      <div class=\\"figcaption\\">
        <p class=\\"small\\">
          test
          
          
        </p>
      </div>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.mobi creates the markup 21`] = `
"
    
      <div class=\\"figure__large figure__large--media figure__inline--square break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.mobi creates the markup 22`] = `
"
    
      <div class=\\"figure__large figure__large--media figure__inline--square break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.mobi creates the markup 23`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--square break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"square\\" alt=\\"Media Fallback Image\\" src=\\"../images/test.jpg\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">Your device does not support embedded media.</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.mobi creates the markup 24`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--square break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"square\\" alt=\\"Media Fallback Image\\" src=\\"../images/test.jpg\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">Your device does not support embedded media.</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;
